modules/notifications/api.php

```diff
@@ -0,0 +1,169 @@
+		}
+
+		return $notifications;
+	}
+
+	private static function fetch_data() : array {
+		$response = wp_remote_get( self::NOTIFICATIONS_URL );
+
+		if ( is_wp_error( $response ) ) {
+			return [];
+		}
+
+		$data = json_decode( wp_remote_retrieve_body( $response ), true );
+
+		if ( empty( $data['notifications'] ) || ! is_array( $data['notifications'] ) ) {
+			return [];
+		}
+
+		return $data['notifications'];
+	}
+
+	private static function add_to_array( $filtered_notifications, $notification ) {
+		foreach ( $filtered_notifications as $filtered_notification ) {
+			if ( $filtered_notification['id'] === $notification['id'] ) {
+				return $filtered_notifications;
+			}
+		}
+
+		$filtered_notifications[] = $notification;
+
+		return $filtered_notifications;
+	}
+
+	private static function check_conditions( $groups ) {
+		foreach ( $groups as $group ) {
+			if ( static::check_group( $group ) ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private static function check_group( $group ) {
+		$is_or_relation = ! empty( $group['relation'] ) && 'OR' === $group['relation'];
+		unset( $group['relation'] );
+		$result = false;
+
+		foreach ( $group as $condition ) {
+			// Reset results for each condition.
+			$result = false;
+			switch ( $condition['type'] ) {
+				case 'wordpress': // phpcs:ignore WordPress.WP.CapitalPDangit.Misspelled
+					// include an unmodified $wp_version
+					include ABSPATH . WPINC . '/version.php';
+					$result = version_compare( $wp_version, $condition['version'], $condition['operator'] );
+					break;
+				case 'multisite':
+					$result = is_multisite() === $condition['multisite'];
+					break;
+				case 'language':
+					$in_array = in_array( get_locale(), $condition['languages'], true );
+					$result = 'in' === $condition['operator'] ? $in_array : ! $in_array;
+					break;
+				case 'plugin':
+					if ( ! empty( $condition['plugin_file'] ) ) {
+						$plugin_file = $condition['plugin_file']; // For PHP Unit tests.
+					} else {
+						$plugin_file = WP_PLUGIN_DIR . '/' . $condition['plugin']; // Default.
+					}
+
+					$version = '';
+
+					if ( is_plugin_active( $condition['plugin'] ) && file_exists( $plugin_file ) ) {
+						$plugin_data = get_plugin_data( $plugin_file );
+						if ( isset( $plugin_data['Version'] ) ) {
+							$version = $plugin_data['Version'];
+						}
+					}
+
+					$result = version_compare( $version, $condition['version'], $condition['operator'] );
+					break;
+				case 'theme':
+					$theme = wp_get_theme();
+					if ( wp_get_theme()->parent() ) {
+						$theme = wp_get_theme()->parent();
+					}
+
+					if ( $theme->get_template() === $condition['theme'] ) {
+						$version = $theme->version;
+					} else {
+						$version = '';
+					}
+
+					$result = version_compare( $version, $condition['version'], $condition['operator'] );
+					break;
+
+				default:
+					$result = apply_filters( "elementor/notifications/condition/{$condition['type']}", $result, $condition );
```

Comment:
```suggestion
					/**
					 * Filters the notification condition, whether to check the group or not.
					 *
					 * The dynamic portion of the hook name, `$condition['type']`, refers to the condition type.
					 *
					 * @since 3.19.0
					 *
					 * @param bool  $result    Whether to check the group.
					 * @param array $condition Notification condition.
					 */
					$result = apply_filters( "elementor/notifications/condition/{$condition['type']}", $result, $condition );
```